code/workspaces/infrastructure-api/src/controllers/projectsController.js

```diff
@@ -47,9 +47,9 @@ async function createOrUpdateProject(request, response, next) {
   const { projectKey } = matchedData(request, { locations: ['params'] });
   const project = matchedData(request, { locations: ['body'] });
 
-  if (projectKey !== project.projectKey) {
+  if (projectKey !== project.key) {
```

Comment:
Could this check be done as a custom validator? Then it could be pushed out of the controller.
https://express-validator.github.io/docs/custom-validators-sanitizers.html

Reply:
That would be nice to do. I will have a look into it.